components/ReviewTransactionModal.tsx

```diff
@@ -0,0 +1,154 @@
+import useNotifications from "@/hooks/useNotifications";
+import useRailgunTx from "@/hooks/useRailgunTx";
+import useShieldPrivateKey from "@/hooks/useShieldPrivateKey";
+import { shortenAddress } from "@/utils/address";
+import { feePercent } from "@/utils/constants";
+import { Button } from "@chakra-ui/button";
+import { Heading, Flex, Box, Text } from "@chakra-ui/layout";
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+} from "@chakra-ui/modal";
+import { Tooltip } from "@chakra-ui/tooltip";
+import {
+  gasEstimateForShield,
+  populateShield,
+} from "@railgun-community/quickstart";
+import {
+  RailgunERC20AmountRecipient,
+  NetworkName,
+  TransactionGasDetailsSerialized,
+  EVMGasType,
+  NETWORK_CONFIG,
+  deserializeTransaction,
+} from "@railgun-community/shared-models";
+import { ethers, constants, BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils.js";
+import { useAccount, useProvider, useSigner } from "wagmi";
+
+type ReviewTransactionModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  tokenAddress?: string;
+  recipient: string;
+  tokenAmount?: string;
+  tokenDecimals?: number;
+  tokenSymbol: string;
+  tokenName: string;
+};
+
+const ReviewTransactionModal = ({
+  isOpen,
+  onClose,
+  tokenAddress,
+  recipient,
+  tokenAmount,
+  tokenDecimals,
+  tokenSymbol,
+  tokenName,
+}: ReviewTransactionModalProps) => {
+  const network = NetworkName.EthereumGoerli; // TODO: Remove once we support multiple chains
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const { isConnected, address } = useAccount();
+  const provider = useProvider();
+  const { data: signer } = useSigner();
+  const { txNotify } = useNotifications();
+  const { shield, isShielding } = useRailgunTx();
+  const bigNumberAmount = parseUnits(tokenAmount! || "0", tokenDecimals);
+  const feeAmount = parseUnits(tokenAmount! || "0", tokenDecimals).div("400");
+  const total = feeAmount.add(bigNumberAmount);
+
+  const doSubmit: React.FormEventHandler = async (e) => {
+    // TODO: Form validation
+    if (!tokenAddress || !tokenAmount || !tokenDecimals || !recipient)
+      throw new Error("bad form");
+    const tx = await shield({
+      tokenAddress: tokenAddress,
+      tokenAmount: tokenAmount,
+      tokenDecimals: tokenDecimals,
+      recipient: recipient,
+    });
```

Comment:
Nit, unless there's any objections, let's use the abbreviated form since key and value names match
```suggestion
    const tx = await shield({ tokenAddress, tokenAmount, tokenDecimals, recipient });
```